packages/core/src/interpreter.ts

```diff
@@ -54,10 +55,11 @@ import { isInFinalState } from './stateUtils';
 import { registry } from './registry';
 import { registerService } from './devTools';
 
-export type StateListener<TContext, TEvent extends EventObject> = (
-  state: State<TContext, TEvent>,
-  event: TEvent
-) => void;
+export type StateListener<
+  TContext,
+  TEvent extends EventObject,
+  TState extends Typestate<TContext> = any
```

Comment:
Could we call this here (and at other places) `TTypestate` or something? `TState` can get easily be interpreted as something different as the word state gets used quite a lot in the codebase 😉 but on the other hand, Typestate is quite a distinct name that would give a better notion of what this holds while reading usage sites of this thing